handbook/engineering/web/goals.md

```diff
@@ -4,29 +4,44 @@
-   Combining this with writing a few smaller extensions (part of goal (3)) allows us to dogfood the experience and inform us where the platform is lacking.
-3. **Build out compelling use cases with the extensions platform.**
-   This includes writing more extensions ourselves, but also extending the extension API with more capabilities to enable more use cases.
-   We have a long list of ideas and use cases customers shared with us that we can solve and/or enable customers to solve with extensions.
-   Some of these integrate Sourcegraph with more external services to suit the setups of more customers, others provide completely new value no other product provides.
+_To reach our long-term goal, we set the following medium-term goals to guide our short-term iteration plans. These medium-term goals are listed in order of rough priority. This means we preference a sooner-listed goal when making progress on one would conflict with another goal, but it doesn't mean we only work on the first goal until it's done – we balance our iterations in sum so we can make some progress on all of these goals over a quarter. Within each goal, the sub-goals are also listed in (stricter) priority order._ 
+
+1. **Make the Sourcegraph web interfaces more consistent and improve discoverability of Sourcegraph features.**
+   1. Make progress on design debt throughout the web app where the web team owns the design: repo page areas and panels, navigation and headers, general theming, and the blob page (file viewer). 
+   1. Build and update designs to help users discover or understand everything Sourcegraph offers, like new navigation menus and header concepts. 
+
+1. **Make Sourcegraph extensions a core part of Sourcegraph users' experiences.**
+   1. Track anonymized, general usage of extensions to determine which extensions are most successful at adding value for our users to inform our future work.
+   1. Build new API endpoints that are robust and obviously useful (used immediately in an extension), to grow adoption of extensions. 
+   1. Make the extensions action bar clear and scalable, and help users discover and use our extensions. 
+   1. Improve discoverability and design of the extension registry to enable users to find the most useful extensions and make users excited to build their own extension. 
+
+1. **Make code insights an entirely new reason to use Sourcegraph.**
+   1. Focus on building prototypes to solve the most pressing needs of our customers, to clearly demonstrate the value of code insights. 
+   1. Expose more generalizable metrics that let our users measure and track their own goals, whether those are migrations, code smells, security needs, cross-collaboration, or other information about code. 
+   1. Expose features that let our users build their own insights. 
+
+1. **Increase the weekly active users of all our code host integrations.**
```

Comment:
@Joelkw the code host integrations are basically just implementations of the extension API (other than the web app, which has a lot of other features). How do you think about support of the new extension API functionlities in code host integrations? For example, we would be faced with the decision of whether we implement the directory decorations only for the webapp, or also on code hosts.

Reply:
Ooh, good question! I can think of some product strategy considerations in this decision, and for now (now = this iteration and the one after it, at least), I'm thinking we leave things specifically like directory decoration out of the code hosts, for the following reasons: 

- I assume the overhead for adding these to code host integrations is "not none" now, and also "not none" in the future. We'd have to build at where to append code within the structure of each code host's directory pages, which means additional overhead that scales linearly with more code hosts. It also means that it could break on a code host if they changed their DOM structure on directory views enough, right? Which means more maintenance overhead as well. 

- I think that the value of our "bring sourcegraph to your code host" integration is mostly around the PR review use case right now, and sometimes in the "googled for a library, found a github/personal website link to a github repo, now I'm browsing the code and sourcegraph.com code intel is cool." In short, around looking at lines of code. I don't think our core value here is "see codecov on the github file view" (and if I'm wrong, we'll hear this from users in feedback like "I want directory decoration on the code host also!" It's a new feature, so we can just truthfully explain we hadn't built it yet - it's not going to break anyone's current workflows). 

- I see advantages to establishing Sourcegraph as the sole place to view code insights, especially as the product is getting started, because it keeps our iteration speed up and feedback loops smaller. At the very least, I see advantages of moving people to "insights viewers" that are not just their code host (Sourcegraph, maybe Backstage, custom BI, anything) because Sourcegraph is best positioned to support UIs built for insights rather than try and fit these insights into an existing page on github. 

Note that these are points specifically for directory decorations. If we were building, say, modals to message someone on slack who wrote the code, those could make more sense to put in the integration. Because this is a case-by-case basis, I think this fits into "Maintain the existing native integrations and browser extensions" because as we build new extensions endpoints, if we decide to put them into the integrations, that is basically now "maintenance" for an existing feature across all our delivery methods.